Feat/univ3 multisig

MisterMard opened the pull request.

MisterMard committed:
fix: impersonate function

MisterMard committed:
add velo usdc-tusd strategy

MisterMard committed:
add prettier rules for typescript

MisterMard committed:
optimism chainlink keeper

MisterMard committed:
add multiple strats to keeper watchlist at once

better watchlist inspection

MisterMard committed:
add beetx ib-reth strategy

MisterMard committed:
fix: fix strategist address for v3 keeper test

MisterMard committed:
fix: chainlinkKeeper test typo

MisterMard committed:
Minichef proxy contract

MisterMard committed:
chef proxy test cleanup

MisterMard committed:
INITIAL

MisterMard committed:
fix: accurate univ3 rebalances

treasury fees in native

setTokenToNativeRoute

single univ3 base for all chains

MisterMard committed:
remove duplicate polygon univ3 bases and interfaces

larrythecucumber321 commented on src/strategies/uniswapv3/strategy-univ3-rebalance.sol:
```diff
@@ -464,50 +429,86 @@ abstract contract StrategyRebalanceUniV3 {
-                _cache.amount1
-            );
-        }
+        bool _zeroForOne = _cache.amount0Desired > _cache.amount0 ? true : false;
```
I know you inherited this naming convention, but it's so confusing that `amount0Desired` is what you already have, and `amount0` is what you desire to obtain 😅

MisterMard committed:
fix: less ambiguous cache variable names

move polygon univ3 migration test to polygon dir

MisterMard committed:
fix: more accurate rebalance implementation

MisterMard committed:
fix: univ3 jar and strategy uses the correct calculation for users deposits and shares

MisterMard committed:
chore: duplicate v3 jar so it can compile, root src requires much cleanup to compile properly

MisterMard committed:
fix: arbitrum univ3 strategies

add tests for arbitrum univ3 migrations

chore: cleanup obsolete files

update safe-core-sdk to latest version to support safes on arbitrum

MisterMard committed:
fix: optimism univ3 strategies

add tests for optimism univ3 migrations

chore: cleanup obsolete files

MisterMard committed:
fix: mainnet univ3 strategies

add tests for mainnet univ3 migrations

fix: multisigHelper ensures owner balance covers txn const

MisterMard committed:
adjust the default performance fees for univ3 strategies to 20%

MisterMard committed:
fix: polygon univ3 strategies

add tests for polygon univ3 migrations

MisterMard committed:
add deployer script files

MisterMard committed:
add v3 migration script

MisterMard committed:
fix: typos

MisterMard committed:
fix: univ3 jar users shares calculation upon deposits

MisterMard committed:
restructure strategy-base-v2

MisterMard committed:
remove problematic jar converters tests

MisterMard committed:
add new velo strategies

update velo unit tests

MisterMard committed:
add base strategy for sushi bento pools

MisterMard committed:
add base unit test for sushi strategies

add fantom sushi strategies and tests

MisterMard committed:
add arbitrum sushi strategies and tests

remove discontinued arbitrum sushi strategies

MisterMard committed:
add polygon sushi strategies and tests

remove discontinued polygon sushi strategies

MisterMard committed:
add kava sushi strategies and tests

MisterMard committed:
add optimism sushi strategies and tests

MisterMard committed:
delete optimism v3 jar code

MisterMard committed:
fix: saddle d4 strategy swap native through a better route
